client/connection.h

```diff
@@ -93,6 +103,21 @@ class LockedConnection final
 		return *ptr;
 	}
 
+	CLMappedBuffer& registerBuffer(IDType id)
+	{
+		// std::map initialises a new object in the map, which is what we want.
+		std::unique_ptr<CLMappedBuffer> buf(new CLMappedBuffer(id));
+		CLMappedBuffer* ptr = buf.get();
+		mParent.mMappedBuffers.insert({id, ptr});
+		buf.release();
+		return *ptr;
+	}
+
+	void unregisterBuffer(IDType id)
+	{
+		mParent.mMappedBuffers.erase(id);
+	}
```

Comment:
Could also set the pointer returned by getBuffer() to nullptr, but that still leaves a stale object in the map (the key will remain).